### ISSUE TITLE
Chapter 4 Blending depending on up/downess of the y coordinate for InOneWeekend

### ISSUE THREAD

**Issue by collinstevens** [RayTracing/raytracing.github.io#306]
In Chapter 4 of **In One Weekend**, it states `The ray_color(ray) function linearly blends white and blue depending on the up/downess of the y coordinate.`, but I was seeing more white near the center of the picture. I assumed it should be the same color for each row because it was described as a `simple gradient`. I realized the statement misleading because the length of the ray cast from the camera to the pixel changes as x changes which changes the normalized vector component y of the direction vector. A note in the text stating as such would have cleared this up, simply stating it is affected by the length of the ray cast and driven off the normalized y component.

**Comment by hollasch** [comment]
Excellent point. I believe that for this example, the simple projected Y value is sufficient, and there's no call for normalizing the length of the ray direction vector.

I'll play around with this.

**Comment by hollasch** [comment]
If you have the code up and running, would you mind yanking out the `unit_vector` call and checking the result?

**Comment by collinstevens** [comment]
Ok, it looks better to me. Although, t is now no longer bounded to -1/+1 correct? At `j = 99`, `v = .99` which would make `vertical = <0.0, 1.98, 0.0>` then `t = 0.5 * (1.98 + 1) = 1.49`.

![before](https://user-images.githubusercontent.com/19230602/70684369-d71f2b80-1c5a-11ea-943e-18ea8ebd4137.png)
![after](https://user-images.githubusercontent.com/19230602/70684374-d8e8ef00-1c5a-11ea-8aa7-9ae7de2b6993.png)

**Comment by hollasch** [comment]
Ah, I see.

Since we're using `ray_color` as a visualization of the Y-axis sweep of the camera ray, perhaps in `ray_color` we should just map the vertical angle to [0,1]. Something like this:

````
vec3 ray_color(const ray& r) {
    auto d = r.direction();
    auto t = 0.5 + atan(r.y() / fabs(r.z()));
    return (1.0-t)*vec3(1.0, 1.0, 1.0) + t*vec3(0.5, 0.7, 1.0);
}
```

**Comment by collinstevens** [comment]
I think you meant to multiply the angle by **0.5**, not add by **0.5**. Nevertheless, that won't work because you'll hit negative angles once you get past the middle of the picture. In my updated comment above, **y** is mapped to [-1, 1] and **t** is mapped to [0, 1] with the _trick_ mentioned in the book.

```
vec3 ray_color(const ray& r) {
	auto direction = r.direction();
	auto t = 0.5 * (direction.y() + 1.0);
	return (1.0 - t) * vec3(1.0, 1.0, 1.0) + t * vec3(0.5, 0.7, 1.0);
}
```

**Comment by hollasch** [comment]
Sorry for the wild goose chase. I finally whipped out the code. This is what I think we should go with:

    vec3 ray_color(const ray& r) {
        auto d = r.direction();
        auto t = 0.5 + atan(d.y() / fabs(d.z())) / pi;
        return lerp(t, vec3(1., 1., 1.), vec3(0., 0.2, 1));
    }

Result:
![image](https://user-images.githubusercontent.com/313871/70687305-9f68b180-1c63-11ea-9eea-6568809c1100.png)

**Comment by hollasch** [comment]
Hmmm. A final simplifying tweak. Let's just use Y component of the direction and call it a day. No need for `atan`.

    auto d = r.direction();
    auto t = 0.5 + 0.5 * d.y();
    return (1.0 - t) * vec3(1., 1., 1.) + t * vec3(0., 0.2, 1);

![image](https://user-images.githubusercontent.com/313871/70687978-82cd7900-1c65-11ea-8f28-584d5eaee0b8.png)

**Comment by collinstevens** [comment]
This makes sense, but we lose the white to blue gradient for a light blue to blue gradient. Our vertical y-range is [-1, 1] and z is always 1 which means the range of our atan is limited further to [-π/4,+π/4]. Applying the / π results in [-1/4, +1/4] and then + 0.5 shifts us to [+1/4, +3/4] so we won't see the white.

**Comment by hollasch** [comment]
ARG! I thought this was a one-off test image. Turns out that this is the background used for all of the first book. Changing the function means changing all subsequent images in the book(s). Ain't nobody got time for that!

I'll stick with the current implementation and instead add clarifying text about the behavior.

**Comment by hollasch** [comment]
Whew. Resolved in #309, development branch.
